tests/test_repos.py

```diff
@@ -0,0 +1,118 @@
+from typing import Generator
+
+import pytest
+from sqlmodel import Session, SQLModel, create_engine
+
+from src.snipster.exceptions import SnippetNotFoundError
+from src.snipster.models import Language, SnippetCreate, SnippetPublic
+from src.snipster.repo import DatabaseBackedSnippetRepo, InMemorySnippetRepo
+
+
+@pytest.fixture
+def snippet() -> SnippetCreate:
+    return SnippetCreate(
+        title="My Snippet", code="print('stuff')", language=Language.PYTHON
+    )
+
+
+@pytest.fixture
+def another_snippet() -> SnippetCreate:
+    return SnippetCreate(
+        title="My Snippet", code="console.log('stuff')", language=Language.JAVASCRIPT
+    )
+
+
+@pytest.fixture(scope="function")
+def im_repo() -> InMemorySnippetRepo:
+    return InMemorySnippetRepo()
+
+
+@pytest.fixture(scope="function")
+def db_repo() -> Generator[DatabaseBackedSnippetRepo, None, None]:
```

Comment:
I see you have a `get_session` in test_models as well, maybe you could centralize some of that in conftest.py (fixtures there get autoloaded by pytest)